codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/SockJsConfig.java

```diff
@@ -0,0 +1,372 @@
+            this.sockJsUrl = sockJsUrl;
+            return this;
+        }
+
+        /**
+         * The session time out in milliseconds.
+         *
+         * @return {@code long} the timeout in ms. The default is {@value #DEFAULT_SESSION_TIMEOUT} ms.
+         */
+        public Builder sessionTimeout(final long ms) {
+            sessionTimeout = ms;
+            return this;
+        }
+
+        /**
+         * Specifies a heartbeat interval.
+         *
+         * @param ms how often that a heartbeat should be sent. Default is {@value #DEFAULT_HEARTBEAT_INTERVAL} ms.
+         */
+        public Builder heartbeatInterval(final long ms) {
+            heartbeatInterval = ms;
+            return this;
+        }
+
+        /**
+         * The max number of types that a streaming transport protocol should allow to be returned
+         * before closing the connection, forcing the client to reconnect. This is done so that the
+         * responseText in the XHR Object will not grow and be come an issue for the client. Instead,
+         * by forcing a reconnect the client will create a new XHR object and this can be see as a
+         * form of garbage collection.
+         *
+         * @param max the max number of bytes that can be written.
+         *            Default is {@value #DEFAULT_MAX_STREAMING_BYTES_SIZE}.
+         */
+        public Builder maxStreamingBytesSize(final int max) {
+            maxStreamingBytesSize = max;
+            return this;
+        }
+
+        /**
+         * Determines whether transport layer security (TLS) should be used.
+         *
+         * @param tls if transport layer security should be used.
+         */
+        public Builder tls(final boolean tls) {
+            this.tls = tls;
+            return this;
+        }
+
+        /**
+         * Specifies the keystore to be used if transport layer security (TLS) is enabled.
+         *
+         * @param keyStore the keystore to be used when TLS is enabled.
+         */
+        public Builder keyStore(final String keyStore) {
+            this.keyStore = keyStore;
+            return this;
+        }
+
+        /**
+         * Specifies the keystore password to be used if transport layer security (TLS) is enabled.
+         *
+         * @param password the keystore password to be used when TLS is enabled.
+         */
+        public Builder keyStorePassword(final String password) {
+            keyStorePassword = password;
+            return this;
+        }
+
+        /**
+         * Builds Config with the previously set values.
+         *
+         * @return {@link SockJsConfig} the configuration for the SockJS service.
+         */
+        public SockJsConfig build() {
+            if (tls && (keyStore == null || keyStorePassword == null)) {
+                throw new IllegalStateException("keyStore and keyStorePassword must be specified if 'tls' is enabled");
+            }
+            return new SockJsConfig(this);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)